test/univariates.jl

```diff
@@ -62,6 +62,13 @@ function verify_and_test(D::Union{Type,Function}, d::UnivariateDistribution, dct
     # test various constructors for promotion, all-Integer args, etc.
     pars = params(d)
 
+    # verify parameter type
+    # truncated parameters may be nothing: Union{Nothing, promote_type()}
+    # but partype should still be type of the non-nothing ones
+    #@test partype(d) === promote_type(typeof.(pars)...)
+    @test partype(d) === promote_type(typeof.(pars)...) || 
+      Union{Nothing,partype(d)} ===  promote_type(typeof.(pars)...)
```

Comment:
Maybe just use something like
```suggestion
    # truncated parameters may be nothing: Union{Nothing, promote_type()}
    # but partype should still be type of the non-nothing ones
    @test partype(d) === mapfoldl(typeof, (S, T) -> T <: Real ? promote_type(S, T) : S, pars; init = Bool)
```

Reply:
Had to wrap my head around, but implemented and tested this more general implementation with an own commit.